docs/reference/type-system.md

```diff
@@ -0,0 +1,27 @@
+# Type System
+
+## Motivation
+
+Feast uses an internal type system to provide guarantees on training and serving data.
+Feast currently supports eight primitive types - `INT32`, `INT64`, `FLOAT32`, `FLOAT64`, `STRING`, `BYTES`, `BOOL`, and `UNIX_TIMESTAMP` - and the corresponding array types.
+The type system is controlled by [`Values.proto`](https://github.com/feast-dev/feast/blob/master/protos/feast/types/Value.proto) in protobuf and by [`types.py`](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/types.py) in Python.
+Type conversion logic can be found in [`type_map.py`](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/type_map.py).
+
+## Examples
```

Comment:
Maybe add a section on getting historical features, and another on getting online features?

Reply:
updated